src/test/java/org/broadinstitute/hellbender/engine/dataflow/ReadsSourceTest.java

```diff
@@ -40,4 +43,32 @@ public void testGetReadPCollectionLocal(){
         DataflowAssert.thatSingleton(count).isEqualTo(7L);
         p.run();
     }
+
+    @Test
+    public void testGetInvalidPCollectionLocal() {
+        // ValidationStringency.SILENT should prevent any read error even though the input has what looks like invalid reads.
+        Pipeline p = TestPipeline.create();
+        ReadsSource source = new ReadsSource(hiSeqBam, p);
+        SAMFileHeader header = source.getHeader();
+        final SAMSequenceDictionary sequenceDictionary = header.getSequenceDictionary();
+        DataflowWorkarounds.registerGenomicsCoders(p);
+        PCollection<Read> reads = source.getReadPCollection(IntervalUtils.getAllIntervalsForReference(sequenceDictionary), ValidationStringency.SILENT);
+        PCollection<Long> count = reads.apply(Count.globally());
+        DataflowAssert.thatSingleton(count).isEqualTo(1674L);
+        p.run();
+    }
+
+    @Test(expectedExceptions = RuntimeException.class)
+    public void testGetInvalidPCollectionLocalStrict() {
+        // ValidationStringency.STRICT should trigger an error on an invalid file
+        Pipeline p = TestPipeline.create();
+        ReadsSource source = new ReadsSource(hiSeqBam, p);
+        SAMFileHeader header = source.getHeader();
+        final SAMSequenceDictionary sequenceDictionary = header.getSequenceDictionary();
+        DataflowWorkarounds.registerGenomicsCoders(p);
+        PCollection<Read> reads = source.getReadPCollection(IntervalUtils.getAllIntervalsForReference(sequenceDictionary), ValidationStringency.STRICT);
+        PCollection<Long> count = reads.apply(Count.globally());
+        DataflowAssert.thatSingleton(count).isEqualTo(1674L);
```

Comment:
Get rid of this because it's confusing if the test is passing because this assert is failing or because the proper exception is being thrown.